WordPress/Classes/Login/WordPressDotComAuthenticator.swift

```diff
@@ -9,46 +9,176 @@ import Alamofire
 ///
 /// API doc: https://developer.wordpress.com/docs/oauth2/
 struct WordPressDotComAuthenticator {
-    enum Error: Swift.Error {
+    enum SignInContext {
+        // Sign in to the app with a WordPress.com account.
+        // Using this context results in automatically reloading the app to display an site in the account.
+        case `default`
+        // Sign in to an existing account.
+        // This is typically used when the app needs to get a new OAuth token because the current one becomes invalid.
+        case reAuthentication(TaggedManagedObjectID<WPAccount>)
```

Comment:
(nit) 
- It needs only the username
- I think `reauthentication` is a single word

Reply:
Updated in https://github.com/wordpress-mobile/WordPress-iOS/pull/23749/commits/6ea0aacf7c3158981616d7100fb6d26145002c06